features/tag_sets/tag_set_associations.feature

```diff
@@ -14,15 +14,33 @@ Feature: Reviewing tag set associations
   Then I should see "Nominated associations were added"
     And I should not see "don't seem to be associated"
 
-  Scenario: If a nominated tag and its parent are wrangled after approval it should still be possible to associate them
+  Scenario: If a nominated tag is wrangled into its nominated parent after approval, it should be automatically associated with the parent
   Given I nominate and approve fandom "Floobry" and character "Zarrr" in "Nominated Tags"
-    And a canonical character "Zarrr" in fandom "Floobry"
```

Comment:
The reason this test didn't work before comes down to this step. It's defined here:
https://github.com/otwcode/otwarchive/blob/8e03660aaa2b470011109a0af187716bf34e575c/features/step_definitions/tag_steps.rb#L76-L80
If the tags in question exist in the database and aren't already canonical, then calling `where(name: name, canonical: true).first_or_create` will create a new tag with the same name, but with `canonical` set to true. These tags will fail to save, because they have duplicate tag names. But there are no calls to `save!` here, so everything will fail silently instead of erroring. Which makes this step a no-op.